src/mgmt/cli_mgmt.c

```diff
@@ -625,14 +641,51 @@ dc_mgmt_net_cfg(const char *name, crt_init_options_t *crt_info)
 	D_STRNDUP(crt_info->cio_provider, info->provider, DAOS_SYS_INFO_STRING_MAX);
 	if (NULL == crt_info->cio_provider)
 		D_GOTO(cleanup, rc = -DER_NOMEM);
-	D_STRNDUP(crt_info->cio_interface, info->interface, DAOS_SYS_INFO_STRING_MAX);
-	if (NULL == crt_info->cio_interface)
-		D_GOTO(cleanup, rc = -DER_NOMEM);
-	D_STRNDUP(crt_info->cio_domain, info->domain, DAOS_SYS_INFO_STRING_MAX);
-	if (NULL == crt_info->cio_domain)
-		D_GOTO(cleanup, rc = -DER_NOMEM);
 
-	D_INFO("Network interface: %s, Domain: %s\n", info->interface, info->domain);
+	d_getenv_bool("D_DYNAMIC_CTX", &d_dynamic_ctx_g);
+	if (d_dynamic_ctx_g) {
+		int i;
+
+		D_ALLOC(crt_info->cio_interface,
+			DAOS_SYS_INFO_STRING_MAX * resp->n_numa_fabric_interfaces);
+		if (crt_info->cio_interface == NULL)
+			D_GOTO(cleanup, rc = -DER_NOMEM);
+		D_ALLOC(crt_info->cio_domain,
+			DAOS_SYS_INFO_STRING_MAX * resp->n_numa_fabric_interfaces);
+		if (crt_info->cio_domain == NULL)
+			D_GOTO(cleanup, rc = -DER_NOMEM);
+
+		for (i = 0; i < resp->n_numa_fabric_interfaces; i++) {
+			Mgmt__FabricInterfaces *numa_ifaces = resp_g->numa_fabric_interfaces[i];
+			int                     j;
+
+			for (j = 0; j < numa_ifaces->n_ifaces; j++) {
+				if (i != 0 || j != 0) {
+					strcat(crt_info->cio_interface, ",");
```

Comment:
you allocate crt_info->cio_interface and crt_info->cio_domain on line 649 to be of size resp->n_numa_fabric_interfaces * [string_max], however here we can fill at max  (resp->n_numa_fabric_interaces * numa_ifaces->n_ifaces) * [string_max] amount. 
if any numa_ifaces->n_ifaces > 1 then you might overflow the array.

Reply:
indeed. i probably got confused with all the names. will fix.